deltachat-ffi/src/lib.rs

```diff
@@ -1284,90 +1289,101 @@ pub unsafe extern "C" fn dc_chatlist_get_context(
 // dc_chat_t
 
 #[no_mangle]
-pub type dc_chat_t<'a> = dc_chat::Chat<'a>;
+pub type dc_chat_t<'a> = chat::Chat<'a>;
 
 #[no_mangle]
 pub unsafe extern "C" fn dc_chat_unref(chat: *mut dc_chat_t) {
     assert!(!chat.is_null());
 
-    dc_chat::dc_chat_unref(chat)
+    Box::from_raw(chat);
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn dc_chat_get_id(chat: *mut dc_chat_t) -> u32 {
     assert!(!chat.is_null());
+    let chat = &*chat;
 
-    dc_chat::dc_chat_get_id(chat)
+    chat::dc_chat_get_id(chat)
 }
 
 #[no_mangle]
 pub unsafe extern "C" fn dc_chat_get_type(chat: *mut dc_chat_t) -> libc::c_int {
     assert!(!chat.is_null());
+    let chat = &*chat;
```

Comment:
general question: is this line in order to get a non-mut ref of chat?

Reply:
`*chat` returns a `dc_chat_t` struct rather than the original pointer to it.
`&` then takes a reference to this to avoid taking ownership.
IIUC